agent/structs/acl.go

```diff
@@ -150,6 +153,19 @@ type ACLToken struct {
 	// to the ACL datacenter and replicated to others.
 	Local bool
 
+	// ExpirationTime represents the point after which a token should be
+	// considered revoked and is eligible for destruction. The zero value
+	// represents NO expiration.
+	ExpirationTime time.Time `json:",omitempty"`
+
+	// ExpirationTTL is a convenience field for helping set ExpirationTime to a
+	// value of CreateTime+ExpirationTTL. This can only be set during
+	// TokenCreate and is cleared and used to initialize the ExpirationTime
+	// field before being persisted to the state store or raft log.
+	//
+	// This is a string version of a time.Duration like "2m".
+	ExpirationTTL string `json:",omitempty"`
```

Comment:
Why can't this be a time.Duration and do the mapstructure magic similar to CreateTime/ExpirationTime to translate from string to Duration?

Reply:
I suppose it could. I had modeled this on `structs.Session:TTL` which used a string.

Reply:
Will change.